packages/gatsby/src/redux/index.ts

```diff
@@ -28,7 +27,8 @@ export const readState = (): IGatsbyState => {
         if (!state.nodesByType.has(type)) {
           state.nodesByType.set(type, new Map())
         }
-        state.nodesByType.get(type)!.set(node.id, node)
+        // eslint-disable-next-line no-unused-expressions
+        state.nodesByType.get(type)?.set(node.id, node)
```

Comment:
why this part was changed? also the change seems "wrong" - we should be guaranteed that `state.nodesByType.get(type)` exists, because of last 3 lines that create entry if it didn't exist before

Reply:
Yeah, maybe it should. We ban non-null assertions in eslint because of potential runtime errors. TypeScript doesn't use `.set` and `.has` as type guards because of potential errors, but I think you're right that it's safe here. I'll switch it back and add a comment.

Reply:
To workaround it we could have something like this? This should make TS happy?

```
let nodesOfThisType: Map<string, Node> = state.nodesByType.get(type)
if (!nodesOfThisType) {
  nodesOfThisType = new Map()
  state.nodesByType.set(type, nodesOfThisType)
}
nodesOfThisType.set(node.id, node)
```

But it just doesn't seem like it should be part of this PR :P

Reply:
OK, changed

Reply:
Yes, that would work (and is the suggested way of handling these)